### ISSUE TITLE
Add selection boundaries

### ISSUE THREAD

**Issue by bjalder26** [ArnoldSmith86/virtualtabletop#390]
I think adding the ability to select multiple widgets at once by using a "selection boundary" (see below) would be a nice feature.  I am not sure if the way this person worked it out would also work here though.  I also worry about what selecting multiple buttons would do, so I imagine if this was deemed a good idea, then it might require a shortcut key to activate it.

![image](https://user-images.githubusercontent.com/78324099/114736878-1ad0ec00-9d0c-11eb-964f-bd0a4ed7d2b8.png)

https://simonwep.github.io/selection/
https://github.com/Simonwep/selection

**Comment by robartsd** [comment]
We might want to limit multiple selection to widgets that are movable (in current context). We also might want to give game designers additional control over what can be part of a multiple selection.

Selected widgets would need to be styled to show the selection to the player who selected them (but not to other players), so we'd probably want to add a class to the DOM but not the shared STATE. The class could be styled with a box-shadow glow.

**Comment by bjalder26** [comment]
> We might want to limit multiple selection to widgets that are movable (in current context). We also might want to give game designers additional control over what can be part of a multiple selection.

Could it be possible to select multiple widgets that aren't movable then click a button to change properties on them?  I guess that would mean the selection would need to "automatically" go into a collection.  If so, then I would prefer a separate property such as "selectable": true, so you could mass select and change properties on widgets that aren't set to movable.  However, if not, then I can't think of a way using a second property (like "selectable") would improve anything.

**Comment by ArnoldSmith86** [comment]
Personally I want something like that in edit mode but I feel like it's not necessary for playing. I'm curious what kind of examples you have though.

**Comment by bjalder26** [comment]
> Personally I want something like that in edit mode but I feel like it's not necessary for playing. I'm curious what kind of examples you have though.

You might be in the process of making such things obsolete, but I wanted to create some in-game tools to make creating maps for RPGs easy.  I've made some progress on that front.  I can create a grid of holders of any size, any dimension, then I can color them to create a fog of war effect (which could disappear as tokens move into new areas).  I can also scoot around a widget to set properties on those holders.  If/When onEnter triggers events, then I believe I can have it so that it will allow someone to import a map, set areas to trigger events, set areas as barriers, set areas as locks, set collectibles in areas, and set up events (text, battles, enemies appearing, characters appearing, etc) without manually editing the JSON at all.  I believe I will also be able to trigger a movement and a new map appearing which will give the impression that someone just moved to a new map (going upstairs, or reaching the left side of a map and entering the right side of another map)

Being able to select a bunch of widgets using a cursor would be easier than button clicks to scoot a button around, but it is not bad as is.  I know that some of the ideas I get are kind of unusual, and I don't expect VTT to pander to me, but sometimes I think there might be other applications that other people will think up.  So I defer to your judgments for actual decisions.

However, if you want to take a peak at what I started doing, then it is here: https://virtualtabletop.io/h3gz
It is a real work in progress though.
